src/client/metadataApiDeploy.ts

```diff
@@ -312,19 +313,11 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
 
     await new Promise((resolve, reject) => {
       // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access,no-underscore-dangle
-      connection.metadata
+      void connection.metadata
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore _invoke is private on the jsforce metadata object, and cancelDeploy is not an exposed method
         ._invoke('cancelDeploy', { id: this.id })
```

Comment:
please make a WI to 
1. add a proper `cancel` method to the metadata class
2. update this code to use it instead of this nastiness